app/models/patient.rb

```diff
@@ -51,18 +51,35 @@ class Patient < ApplicationRecord
 
   attribute :call_result, :string
 
-  #
-  # Note: This scope expects a join(:blood_pressures) to exist.
-  # For eg, Patient.joins(:blood_pressures).follow_ups(:month).
-  #
-  # It doesn't include the join in this scope to play well with certain parent scopes (eg. Facility).
-  # Parent scopes might auto add this join and the final query would end up with unnecessary joins, affecting perf.
-  #
-  scope :follow_ups, -> (period, last: nil) {
-    where("patients.recorded_at < #{BloodPressure.date_to_period_sql(period)}")
-      .group_by_period(period, 'blood_pressures.recorded_at', last: last)
+  scope :with_diabetes, -> { joins(:medical_history).merge(MedicalHistory.diabetes_yes) }
+  scope :with_hypertension, -> { joins(:medical_history).merge(MedicalHistory.hypertension_yes) }
+
+  scope :follow_ups_by_period, -> (period, last: nil) {
+    follow_ups_with(Encounter, period, time_column: 'encountered_on', last: last)
+  }
+
+  scope :diabetes_follow_ups_by_period, -> (period, last: nil) {
+    follow_ups_with(BloodSugar, period, last: last)
+      .with_diabetes
+  }
+
+  scope :hypertension_follow_ups_by_period, -> (period, last: nil) {
+    follow_ups_with(BloodPressure, period, last: last)
+      .with_hypertension
   }
 
+  def self.follow_ups_with(model_name, period, time_column: 'recorded_at', last: nil)
+    table_name = model_name.table_name.to_sym
```

Comment:
Do we need `to_sym`? Is it just being `to_s`'d below anyway?

Reply:
It's for the `joins`. If you pass in a string to the `joins` it doesn't scope it against the class correctly. I can `to_sym` in the `joins` if that's better? Don't have a particular preference. It's easier to begin with symbols since they seem to have a special meaning instead of remembering to `to_sym` everywhere.

Reply:
Meh, was just looking for cruft. This is used 👍